control-plane/csi-controller/src/main.rs

```diff
@@ -1,55 +1,82 @@
-use env_logger::{Builder, Env};
-use git_version::git_version;
-use structopt::StructOpt;
+use tracing::info;
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt, EnvFilter, Registry};
+
+use clap::{App, Arg};
 
 mod client;
-pub mod controller;
-pub mod identity;
-pub use client::{ApiClientError, MayastorApiClient};
+mod controller;
+mod identity;
+use client::{ApiClientError, MayastorApiClient};
 mod server;
 
-#[macro_use]
-extern crate log;
-extern crate env_logger;
-
 const CSI_SOCKET: &str = "/var/tmp/csi.sock";
 
-#[derive(Debug, Clone, StructOpt)]
-#[structopt(
-    name = "Mayastor CSI Controller",
-    about = "test",
-    version = git_version!(args = ["--tags", "--abbrev=12"], fallback="unkown"),
-    setting(structopt::clap::AppSettings::ColoredHelp)
-)]
-
-struct CsiControllerCliArgs {
-    #[structopt(short = "r", long = "rest-endpoint")]
-    pub rest_endpoint: String,
-    #[structopt(short="c", long="csi-socket", default_value=CSI_SOCKET)]
-    pub csi_socket: String,
-    #[structopt(short="l", long="log-level", default_value="info", possible_values=&["info", "debug", "trace"])]
-    pub log_level: String,
-}
+#[tokio::main(worker_threads = 2)]
+pub async fn main() -> Result<(), String> {
+    let args = App::new("Mayastor k8s pool operator")
```

Comment:
any particular reason for replacing structopt with clap, we tend to use struct on most of the control plane

Reply:
Oh, well -- to be honest, I compared it to the MSP bits. I prefer (and yes, its preference) to be more in and just explicitly type things when it comes to this. More often than not, I'm trying to unravel "what as the attribute again?" then simply typing it out. 

I think I was looking for  "How do say: use this or use env variable"... So no real reason other than that.